src/bosh-softlayer-cpi/softlayer/common/vm_utils.go

```diff
@@ -43,28 +43,34 @@ func CreateVirtualGuestTemplate(stemcell bslcstem.Stemcell, cloudProps VMCloudPr
 		switch network.Type {
 		case "dynamic":
 			if value, ok := network.CloudProperties["PrimaryNetworkComponent"]; ok {
+				networkVlan := sldatatypes.NetworkVlan{}
 				networkComponent := value.(map[string]interface{})
 				if value1, ok := networkComponent["NetworkVlan"]; ok {
-					networkValn := value1.(map[string]interface{})
-					if value2, ok := networkValn["Id"]; ok {
-						cloudProps.PrimaryNetworkComponent = sldatatypes.PrimaryNetworkComponent{
-							NetworkVlan: sldatatypes.NetworkVlan{
-								Id: int(value2.(float64)),
-							},
-						}
+					networkVlanInfo := value1.(map[string]interface{})
+					if value2, ok := networkVlanInfo["Id"]; ok {
+						networkVlan.Id = int(value2.(float64))
+					}
+					if value2, ok := networkVlanInfo["PrimarySubnetId"]; ok {
+						networkVlan.PrimarySubnetId = int(value2.(float64))
+					}
+					cloudProps.PrimaryNetworkComponent = sldatatypes.PrimaryNetworkComponent{
+						NetworkVlan: networkVlan,
 					}
 				}
 			}
 			if value, ok := network.CloudProperties["PrimaryBackendNetworkComponent"]; ok {
```

Comment:
This is pretty deep nesting, so I am counting:
```
case "dynamic""
  if ...
    if ...
      if ...
        if ...
case "xyz":
...
```

Clearly can be simplified. Not more than two or max three level deep. So case then maybe one or max two if statements. Plus using functions you can simplify this so it reads well. Case then call function and in function you can do necessary conditions.